File: rookecomshop-admin/src/types/query-dto.ts
import { ProductStatus } from '@/services/products/products.enum';

export type QueryDto = {
	searchTerm?: string;
	sortOrder?: string;
	sortColumn?: string;
	page: number;
	pageSize: number;
};

export type ProductQueryDto = QueryDto & {
	categoryName: string | null;
	status: ProductStatus | null;
};

export const defaultQuery: QueryDto = {
	page: 1,
	pageSize: 10,
};

File: src/Core/RookEcomShop.Application/Dto/QueryDto.cs
namespace RookEcomShop.Application.Dto
{
    public class QueryDto
    {
        public string? SearchTerm { get; set; }
        public string? SortOrder { get; set; }
        public string? SortColumn { get; set; }
        public int Page { get; set; } = 1;
        public int PageSize { get; set; } = 10;
    }
}